### PR TITLE
Document span lifetime issue in CreateSpan

### PR DESCRIPTION
**Pull request by jaredpar**
The language rules around span safety that C# and F# adhere to assume
there is no way to create a `Span<T>` wrapper over a `ref` local /
parameter. This means `ref` inputs into a method are not considered when
calculating the allowed lifetime of a returned `Span<T>`. Hence both
CreateSpan and CreateReadOnlySpan will be assumed to have heap lifetime
even when provided stack based inputs. Example:

``` c#
Span<int> Example() {
  int i = 42;
  Span<int> span = MemoryMarshal.CreateSpan(ref i, length: 1);
  return span; // C# and F# will allow this
}
```

In this case the actual lifetime of `span` is that of `i`. Yet the
compiler doesn't consider the `ref i` input and hence believes this must
be heap based and hence safe to return out of the method.

This is okay as these methods are unsafe. But want to explicitly
document that fact.

More information on the safety rules can be found in the [span safety
proposal](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.2/span-safety.md)

### CODE REVIEW

File: src/Common/src/CoreLib/System/Runtime/InteropServices/MemoryMarshal.Fast.cs [hunk 1]
```diff
@@ -214,6 +214,7 @@ public static ReadOnlySpan<byte> AsBytes<T>(ReadOnlySpan<T> span)
         /// </summary>
         /// <param name="reference">A reference to data.</param>
         /// <param name="length">The number of <typeparamref name="T"/> elements the memory contains.</param>
+        /// <returns>The lifetime of the returned span will not be validated for safety by span aware languages.</returns>
```
Comment (sharwell) [inline src/Common/src/CoreLib/System/Runtime/InteropServices/MemoryMarshal.Fast.cs]: :bulb: span-aware

Comment (sharwell) [reply src/Common/src/CoreLib/System/Runtime/InteropServices/MemoryMarshal.Fast.cs]: hyphen-minus

:trollface:

**Comment by jkotas** [comment]
Related to https://github.com/dotnet/docs/issues/4912

**Pull request merged**